rust/src/dns/log.rs

```diff
@@ -584,92 +550,132 @@ fn dns_log_json_answer(
         js_answers.close()?;
 
         if flags & LOG_FORMAT_DETAILED != 0 {
-            js.set_object("answers", &js_answers)?;
+            jb.set_object("answers", &js_answers)?;
         }
 
         if flags & LOG_FORMAT_GROUPED != 0 {
-            js.open_object("grouped")?;
+            jb.open_object("grouped")?;
             for (k, mut v) in answer_types.drain() {
                 v.close()?;
-                js.set_object(&k, &v)?;
+                jb.set_object(&k, &v)?;
             }
-            js.close()?;
-        }
-    }
-
-    if !response.authorities.is_empty() {
-        js.open_array("authorities")?;
-        for auth in &response.authorities {
-            let auth_detail = dns_log_json_answer_detail(auth)?;
-            js.append_object(&auth_detail)?;
+            jb.close()?;
         }
-        js.close()?;
     }
-
     Ok(())
 }
 
-fn dns_log_query(
-    tx: &DNSTransaction, i: u16, flags: u64, jb: &mut JsonBuilder,
-) -> Result<bool, JsonError> {
-    let index = i as usize;
-    if let Some(request) = &tx.request {
-        if index < request.queries.len() {
-            let query = &request.queries[index];
+/// Common logger for DNS requests and responses.
+///
+/// It is expected that the JsonBuilder is an open object that the DNS
+/// transaction will be logged into. This function will not create the
+/// "dns" object.
+fn log_json(tx: &mut DNSTransaction, flags: u64, jb: &mut JsonBuilder) -> Result<(), JsonError> {
+    jb.set_int("version", 3)?;
+
+    let message = if let Some(request) = &tx.request {
+        jb.set_string("type", "request")?;
+        request
+    } else if let Some(response) = &tx.response {
+        jb.set_string("type", "response")?;
+        response
+    } else {
+        debug_validate_fail!("unreachable");
+        return Ok(());
+    };
+
+    // The internal Suricata transaction ID.
+    jb.set_uint("tx_id", tx.id - 1)?;
+
+    // The on the wire DNS transaction ID.
+    jb.set_uint("id", tx.tx_id() as u64)?;
+
+    // Log header fields. Should this be a sub-object?
```

Comment:
Oops, forgot to do this: https://github.com/OISF/suricata/pull/11238#discussion_r1626682580

Reply:
So, is this PR a draft ?